app/models/notice.rb

```diff
@@ -186,10 +186,10 @@ def self.in_topics(topics)
       .uniq
   end
 
-  def self.submitted_by(submitters)
+  def self.by_role(enitites_involved, role)
```

Comment:
Because you're adding an argument to this function signature, please supply it with a default value (in this case `:submitters`) so callers still work without needing to update their arguments. (And also please fix the spelling on `entities_involved`.)

Reply:
Also think through the values that `role` is permitted to take. What happens if this function gets an unexpected value for `role`? Do we need to take steps to prevent or handle that case? (Steps might include: documenting expectations to make it easier for future developers to use this function; adding a guard clause that checks to see if `role` (or `role.to_sym`) is within an expected set and returning an empty set if not.)

Reply:
> 3. What do you expect this function to be returning? Currently it returns an array of elasticsearch results, which are basically JSON -- which is to say, NOT ActiveRecord instances. It's totally fine to be returning ES and not ActiveRecord as long as that's what you intend to be doing, but if you're trying to act on the data as if it's ActiveRecord when it's not you might be getting errors.

Yes, I think ES data should be fine here, because the search suggestions would only need the entity name and id which can be extracted from the json.

Reply:
👍